src/libraries/Microsoft.PowerFx.Core/Binding/BinderUtils.cs

```diff
@@ -140,7 +140,19 @@ internal static bool TryGetBestOverload(TexlBinding txb, CallNode node, DType[]
                 var localWarnings = new LimitedSeverityErrorContainer(txb.ErrorContainer, DocumentErrorSeverity.Warning);
 
                 // Typecheck the invocation and infer the return type.
-                typeCheckSucceeded = maybeFunc.HandleCheckInvocation(txb, args, argTypes, localWarnings, out returnType, out nodeToCoercedTypeMap);
+                //typeCheckSucceeded = maybeFunc.HandleCheckInvocation(txb, args, argTypes, localWarnings, out returnType, out nodeToCoercedTypeMap);
+
+                typeCheckSucceeded = HandleCheckInvocationWithDeferred(maybeFunc, txb, args, argTypes, out var checkInvocationErrors, out returnType, out nodeToCoercedTypeMap);
+
+                // Adding back all the errors to local warnings
+                foreach (var error in checkInvocationErrors.GetErrors())
```

Comment:
Should this use the record/replay feature bryan added in #798 ?

Reply:
Looks like this is creating 2 layers of throwaway error containers, can it be more efficient?